lib/pbench/server/db/models/tracker.py

```diff
@@ -356,12 +446,14 @@ def advance(self, new_state):
             DatasetBadStateTransition: The dataset does not support transition
                 from the current state to the desired state.
         """
+        if type(new_state) is not States:
+            raise DatasetBadState(f'state value "{new_state!r}" is not a States enum')
```

Comment:
I'm not suggesting that you change this, but, in the future, when printing out a message indicating what something is not, it's often helpful to also point out what it actually is.  (That is, presumably the caller _thought_ it had it right, and so the extra detail can help point to where it went wrong.)